global-news-app/src/Charts/ChartsSection/DatesChart/DatesChart.tsx

```diff
@@ -0,0 +1,108 @@
+import { AreaChart, Area, XAxis } from "recharts";
+import { format } from "date-fns";
+import { datesDataMock } from "Utils/MockData";
+import ChartCard from "../ChartCard/ChartCard";
+import { DatesChartContainer } from "./DatesChart.style";
+import { colors } from "assets/styles/theme";
+import { ChartsProps } from "../ChartsSection";
+import EmptyState from "components/EmptyState/EmptyState";
+import { Charts } from "Types/enums";
+import { isEmpty } from "lodash";
+
+interface IDateCounts {
+  [date: string]: number;
+}
+interface IMonths {
+  date: Date;
+  value: number;
+}
+
+const DatesChart = ({ articles }: ChartsProps) => {
+  const formatedMonthData = () => {
+    const sortFunc = (obj1: IMonths, obj2: IMonths) => {
+      const timestamp1 = obj1.date.getTime();
+      const timestamp2 = obj2.date.getTime();
+      return timestamp1 - timestamp2;
+    };
+
+    const dateCounts: IDateCounts = {};
+
+    articles.forEach((article) => {
+      const month = `${article?.publishedAt?.substring(0, 7)}-01`;
```

Comment:
for the magic nimbers create a const 
`const SEVEN_CHARS = 7`
and then
`const month = `${article?.publishedAt?.substring(0, SEVEN_CHARS)}-01`;`